src/components/app/routes/loaders/rootLoader.js

```diff
@@ -148,16 +148,54 @@ const fetchEnterpriseLearnerData = async (username, options = {}) => {
   const activeLinkedEnterpriseCustomerUser = linkedEnterpriseCustomersUsers.find(enterprise => enterprise.active);
   const activeEnterpriseCustomer = activeLinkedEnterpriseCustomerUser?.enterpriseCustomer;
   const activeEnterpriseCustomerUserRoleAssignments = activeLinkedEnterpriseCustomerUser?.roleAssignments;
+
+  // Find enterprise customer metadata for the currently viewed
+  // enterprise slug in the page route params.
+  const foundEnterpriseCustomerUserForCurrentSlug = linkedEnterpriseCustomersUsers.find(
+    enterpriseCustomerUser => enterpriseCustomerUser.enterpriseCustomer.slug === enterpriseSlug,
+  );
+
+  const determineEnterpriseCustomerUserForDisplay = () => {
+    const activeEnterpriseCustomerUser = {
+      enterpriseCustomer: activeEnterpriseCustomer,
+      roleAssignments: activeEnterpriseCustomerUserRoleAssignments,
+    };
+    if (!enterpriseSlug) {
+      return activeEnterpriseCustomerUser;
+    }
+    if (enterpriseSlug !== activeEnterpriseCustomer.slug && foundEnterpriseCustomerUserForCurrentSlug) {
+      return {
+        enterpriseCustomer: foundEnterpriseCustomerUserForCurrentSlug.enterpriseCustomer,
+        roleAssignments: foundEnterpriseCustomerUserForCurrentSlug.roleAssignments,
+      };
+    }
+    return activeEnterpriseCustomerUser;
+  };
+
+  const {
+    enterpriseCustomer,
+    roleAssignments,
+  } = determineEnterpriseCustomerUserForDisplay();
+  const enterpriseCustomerUserRoleAssignments = roleAssignments;
```

Comment:
[nit] could probably clean this line up by returning 
```js
return {
    enterpriseCustomer,
    enterpriseCustomerRoleAssignments: roleAssignments
    activeEnterpriseCustomer,
    activeEnterpriseCustomerUserRoleAssignments,
    allLinkedEnterpriseCustomerUsers: linkedEnterpriseCustomersUsers
}
```

Reply:
You are correct. Updated!